test_util/src/lib.rs

```diff
@@ -108,44 +123,242 @@ pub fn generate_aead_key() -> LessSafeKey {
     LessSafeKey::new(unbound_key)
 }
 
+#[derive(Debug)]
+struct MockClockInner {
```

Comment:
It looks like a watch channel would work great here, as borrow() doesn't require awaiting on anything.

(I had previously tried using a `std::sync::Mutex` plus a `Condvar`, but that interacted poorly with the runtime. I tried implementing `sleep()` by waiting on the condition variable from within `block_in_place()`, but that went poorly because it resulted in a `Future::poll()` that blocked when polled, and the runtime deadlocked on it.)